src/core/core.element.js

```diff
@@ -3,15 +3,28 @@
 import {extend, inherits} from '../helpers/helpers.core';
 import {isNumber} from '../helpers/helpers.math';
 
+/**
+ * @typedef {import("./core.animation").default} Animation
+ */
+
 class Element {
 
 	/**
-	 * @param {object} [cfg] optional configuration
+	 * @param {{x?: number, y?: number, hidden?: boolean, active?: boolean, options?: object}} [cfg] optional configuration
 	 */
 	constructor(cfg) {
+		/** @type {number} */
```

Comment:
We're close to being able to turn on `strictNullChecks` after https://github.com/chartjs/Chart.js/pull/7085. That will require that you not initialize to `undefined` since the type is declared as `number`, so probably better to set as `-1` here. Though I think I'd just do `this.x = cfg.x` and remove the `extend` call